### PR TITLE
added online help for parameter ok

### PR DESCRIPTION
**Pull request by luciahroyalty101**
Added online help for param ok.

### CODE REVIEW

**Comment by luciahroyalty101** [comment]
Added online help on parameter ok.

**Comment by MarkEWaite** [comment]
Needs review from either @MarkEWaite , @kwhetstone , @aHenryJard , or @stackscribe

File: pom.xml [hunk 1]
```diff
@@ -116,5 +116,9 @@
             <artifactId>credentials-binding</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>ssh-credentials</artifactId>
+        </dependency>
```
Comment (MarkEWaite) [inline pom.xml:122]: This should not be needed.
```suggestion
```

File: src/main/resources/org/jenkinsci/plugins/workflow/support/steps/input/InputStep/help-ok.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<p>
+    Every input step has a timeout especially for cases where there is no responce from the user..Jenkins aborts the job execution automatically.Param ok enables to question and agrees the action by Jenkins to be automated.
+</p>
```
Comment (MarkEWaite) [inline src/main/resources/org/jenkinsci/plugins/workflow/support/steps/input/InputStep/help-ok.html:3]:
```suggestion
<div>
    Provides the text that will be displayed in the "ok" button that allows the Pipeline to continue.
</div>
```

I used the Pipeline snippet generator with these values:

screencapture-mark-pc2-markwaite-net-8080-pipeline-syntax-2021-04-22-13_31_48-edit

to create this pipeline:

```
pipeline {
    agent any

    stages {
        stage('Hello') {
            steps {
                echo 'Hello World'
                script {
                def ret = input   id: 'A-custom-id-for-this-input-step',
                        message: 'This is the message',
                        ok: 'This is the OK button',
                        parameters: [
                            booleanParam(defaultValue: true,
                                         description: 'Boolean parameter to <code>input</code> step',
                                         name: 'A_BOOLEAN_PARAMETER_TO_INPUT')
                        ], 
                        submitter: 'mwaite,alice,bob,carl,devin,edgar',
                        submitterParameter: 'approvingSubmitterParameter'
                    echo "Has submitterParameter = ${ret.approvingSubmitterParameter}"
                    echo "Has A_BOOLEAN_PARAMETER_TO_INPUT = ${ret.A_BOOLEAN_PARAMETER_TO_INPUT}"
                }
                echo 'Goodbye World'
            }
        }
    }
}
```

that generated this result in the Jenkins Blue Ocean UI:

screencapture-mark-pc2-markwaite-net-8080-blue-organizations-jenkins-pipeline-exploring-input-detail-pipeline-exploring-input-1-pipeline-2021-04-22-13_33_24

**Review by MarkEWaite — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (aHenryJard) [reply pom.xml:122]: I agree with @MarkEWaite  you should not need additional dependency.

**Comment by luciahroyalty101** [comment]
Its okay..lemme make the modifiacation.

**Review by MarkEWaite — APPROVED** [review]

**Review by oleg-nenashev — COMMENTED** [review]
Looks like you use Asciidoc formatting in the Markdown document

File: README.md [hunk 3]
```diff
@@ -2,7 +2,11 @@
 [![Jenkins Plugin](https://img.shields.io/jenkins/plugin/v/pipeline-input-step.svg)](https://plugins.jenkins.io/pipeline-input-step)
 [![Jenkins Plugin Installs](https://img.shields.io/jenkins/plugin/i/pipeline-input-step.svg?color=blue)](https://plugins.jenkins.io/pipeline-input-step)
 
-Adds the Pipeline `input` step to wait for human input or approval.
+Adds the Pipeline `input` step to wait for human input or approval. 
+
+For further understanding,check the https://plugins.jenkins.io[documentation] for pipeline-input-step plugin.  The link:https://www.jenkins.io/redirect/pipeline-snippet-generator [Pipeline Syntax Snippet Generator] guides the user to select checkout options.
```
Comment (MarkEWaite) [inline README.md:7]: Use markdown format instead of asciidoc format for the hyperlink markup.
```suggestion
For further understanding,check the [documentation](https://plugins.jenkins.io) for pipeline-input-step plugin.  The  [Pipeline Syntax Snippet Generator](https://www.jenkins.io/redirect/pipeline-snippet-generator) guides the user to select checkout options.
```

File: README.md [hunk 4]
```diff
@@ -2,7 +2,11 @@
...
+
+
 
 ## Version History
-Please refer to [the changelog](CHANGELOG.md)
+Please refer to [the changelog](CHANGELOG.md) SCM link:https://www.jenkins.io/doc/pipeline/steps/workflow-scm-step/[`checkout` step].
```
Comment (MarkEWaite) [inline README.md:12]:
```suggestion
Please refer to [the changelog](CHANGELOG.md) and SCM [`checkout` step](https://www.jenkins.io/doc/pipeline/steps/workflow-scm-step/).
```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]

**Comment by MarkEWaite** [comment]
@MarkEWaite the ci.jenkins.io tests are flaky, could you run it again?

File: README.md [hunk 5]
```diff
@@ -2,7 +2,11 @@
...
-Adds the Pipeline `input` step to wait for human input or approval.
+Adds the Pipeline `input` step to wait for human input or approval. 
+
+For further understanding,check the [documentation](https://plugins.jenkins.io) for pipeline-input-step plugin.  The  [Pipeline Syntax Snippet Generator](https://www.jenkins.io/redirect/pipeline-snippet-generator) guides the user to select checkout options.
+
+
 
 ## Version History
-Please refer to [the changelog](CHANGELOG.md)
+Please refer to [the changelog](CHANGELOG.md) and SCM [`checkout` step](https://www.jenkins.io/doc/pipeline/steps/workflow-scm-step/).
```
Comment (MarkEWaite) [inline README.md:12]: I recommend that this section be left as it was previously, a reference to the version history, rather than mixing a reference to something that is not related to version history.
```suggestion
Please refer to [the changelog](CHANGELOG.md)
```

File: README.md [hunk 6]
```diff
@@ -2,7 +2,11 @@
 [![Jenkins Plugin](https://img.shields.io/jenkins/plugin/v/pipeline-input-step.svg)](https://plugins.jenkins.io/pipeline-input-step)
 [![Jenkins Plugin Installs](https://img.shields.io/jenkins/plugin/i/pipeline-input-step.svg?color=blue)](https://plugins.jenkins.io/pipeline-input-step)
 
-Adds the Pipeline `input` step to wait for human input or approval.
+Adds the Pipeline `input` step to wait for human input or approval. 
+
+For further understanding,check the [documentation](https://plugins.jenkins.io) for pipeline-input-step plugin.  The  [Pipeline Syntax Snippet Generator](https://www.jenkins.io/redirect/pipeline-snippet-generator) guides the user to select checkout options.
```
Comment (MarkEWaite) [inline README.md:7]: I think it would be better if the first sentence of the paragraph recommended the Pipeline Syntax Snippet Generator and the second sentence directed them to the detailed documentation of the input step.  Directing them to the https://plugins.jenkins.io page causes the reader to open a page with very little content.
```suggestion
Use the  [Pipeline Syntax Snippet Generator](https://www.jenkins.io/redirect/pipeline-snippet-generator) to select options for the `input` step.
For further understanding, check the [pipeline input step plugin documentation](https://www.jenkins.io/doc/pipeline/steps/pipeline-input-step/).
```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
Thanks for recommending the Pipeline Syntax Snippet Generator.  I've proposed some changes in ordering and link destinations.  Would those changes be OK for you @luciahroyalty101 ?

**Review by MarkEWaite — APPROVED** [review]

**Review by MRamonLeon — APPROVED** [review]
